cpp/benchmarks/transpose/transpose.cpp

```diff
@@ -40,6 +42,17 @@ static void BM_transpose(benchmark::State& state)
     cuda_event_timer raii(state, true);
     auto output = cudf::transpose(input);
   }
+
+  // Collect memory statistics.
+  auto const bytes_read = static_cast<uint64_t>(input.num_columns()) * input.num_rows() *
+                          sizeof(cudf::id_to_type<column_type_id>);
+  auto const bytes_written = bytes_read;
+  // Account for nullability in input and output.
+  auto const null_bytes =
+    2 * input.num_columns() * cudf::bitmask_allocation_size_bytes(input.num_rows());
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**suggestion:** ‏This one could also overflow, I think, perhaps:
```suggestion
    2 * static_cast<uint64_t>(input.num_columns()) * cudf::bitmask_allocation_size_bytes(input.num_rows());
```
?

Reply:
Are you sure about this one? Since the return type of `cudf::bitmask_allocation_size_bytes` is `std::size_t` which is either `unsigned long` or `unsigned long long` so for reasonable input sizes the integer type promotion will avoid the overflow (https://cppinsights.io/s/26f977cb).

That said, just having this discussion indicates I should've included an explicit cast upfront to clear up any potential confusion.

Reply:
Left-to-right associativity means that this is evaluated as `(2 * ncol) * nrow`, the first multiplication is performed in `size_type` (AKA, `int32_t`), so that could overflow, no? Although I think these benchmarks are generally run with fewer than $2^{30}$ rows, there's in general no reason why they couldn't be (although the transpose performance will be _terrible_ I grant you).

Reply:
Why not just always put the thing that returns `size_t` (`sizeof` or `bitmask_allocation_size_bytes`) first in the arithmetic in all of these PRs?

Reply:
Personally, I would keep the cast explicit to make visible what is happening, but I don't have a strong stance on it.